File: 861.cpp
struct Solution {
    int matrixScore(vector<vector<int>>& grid) {
        int m = grid.size(), n = grid[0].size(), lead, flip;
        vector<int> nums(m);
        for (int i = 0; i < m; ++i){
            lead = 1 << (n - 1);
            for (int j = 0; j < n; ++j){
                nums[i] |= grid[i][j] * lead;
                lead >>= 1;
            }
        }
        lead = 1 << (n - 1);
        flip = (1 << n) - 1;
        for (int &x : nums){
            if ((x & lead) == 0){
                x ^= flip;
            }
        }
        lead = 1;
        for (int j = 0; j < n; ++j){
            flip = 0;
            for (int &x : nums){
                flip += ((x & lead) != 0);
            }
            if (flip <= m / 2){
                for (int &x : nums){
                    x ^= lead;
                }
            }
            lead <<= 1;
        }
        return reduce(nums.begin(), nums.end());
    }
};

File: 861.py
class Solution:
    def matrixScore(self, grid: List[List[int]]) -> int:
        m = len(grid)
        n = len(grid[0])
        nums = [
            sum(row[j] * (1 << (n - 1 - j)) for j in range(n))
            for row in grid
        ]
        lead = 1 << (n - 1)
        flip = (1 << n) - 1
        for i in range(m):
            if nums[i] & lead == 0:
                nums[i] ^= flip
        lead = 1
        for j in range(n):
            if sum((x & lead) // lead for x in nums) <= m // 2:
                for i in range(m):
                    nums[i] ^= lead
            lead <<= 1
        return sum(nums)
